tests/unit_tests/test_mask.py

```diff
@@ -0,0 +1,371 @@
+        ),
+        (  # test different replacement values: replacement = string
+            {
+                "data": data_3bands_3dates,
+                "mask": mask_3bands_3dates_num,
+                "scenes_data": scenes_3dates,
+                "scenes_mask": scenes_3dates,
+                "replacement": replacement_val_str,
+            },
+            None,
+            resultWithReplacement(result_3bands_3dates, replacement_val_str),
+        ),
+        (  # test mask parameter missing dimensions: mask parameter with no temporal dimension
+            {
+                "data": data_3bands_3dates,
+                "mask": [mask_3bands_3dates_num[0]],
+                "scenes_data": scenes_3dates,
+                "scenes_mask": None,
+            },
+            "maskCube.removeDimension('temporal_name');",
+            resultWithReplacement(result_mask_no_temporal),
+        ),
+        (  # test mask parameter missing dimensions: mask parameter with no bands dimension
+            {
+                "data": data_3bands_3dates,
+                "mask": [{"B01": el["B01"]} for el in mask_3bands_3dates_num],
+                "scenes_data": scenes_3dates,
+                "scenes_mask": scenes_3dates,
+            },
+            "maskCube.removeDimension('bands_name');",
+            resultWithReplacement(result_mask_no_bands),
+        ),
+        (  # test mask parameter missing dimensions: data parameter with additional dimension
+            {
+                "data": data_3bands_3dates,
+                "mask": mask_3bands_3dates_num,
+                "scenes_data": scenes_3dates,
+                "scenes_mask": scenes_3dates,
+            },
+            "dataCube.addDimension('test_name', 'test_label', 'other');",
+            result_added_dim,
+        ),
+    ],
+)
+def test_mask(
+    mask_process_code,
+    example_input,
+    additional_js_code_specific_to_case,
+    expected_output,
+):
+
+    data_parameter = json.dumps(example_input["data"])
+    mask_parameter = json.dumps(example_input["mask"])
+    scenes_data = json.dumps(example_input["scenes_data"])
+    scenes_mask = json.dumps(example_input["scenes_mask"])
+    replacement_parameter = (
+        json.dumps(example_input["replacement"])
+        if "replacement" in example_input
+        else None
+    )
+
+    vars_definitions = (
+        f"const dataCube = new DataCube({data_parameter}, 'bands_name', 'temporal_name', true, [], {scenes_data});"
+        + f"let maskCube = new DataCube({mask_parameter}, 'bands_name', 'temporal_name', true, [], {scenes_mask});"
+    )
+
+    if replacement_parameter:
+        vars_definitions = (
+            vars_definitions + f"const replacement = {replacement_parameter};"
+        )
+
+    additional_js_code_to_run = (
+        load_datacube_code()
+        + vars_definitions
+        + (additional_js_code_specific_to_case or "")
+    )
+
+    arguments = f"'data': dataCube, 'mask': maskCube, 'scenes': {scenes_data}"
```

Comment:
I think `scenes` does not need be passed to `mask` process.

Reply:
True, thanks. Removed.